dev-java/batik: add 1.18 | dev-java/fop: add 2.10 | dev-java/jna: add 5.15.0

vaukai opened the pull request:
<!-- Please put the pull request description above -->

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

fordfrog commented:
what's approved is merged

fordfrog commented:
i just merged the drops

fordfrog reviewed (commented):
should really 2.16.1 be dropped instead of 2.15.1?

vaukai commented:
> should really 2.16.1 be dropped instead of 2.15.1?

nah, thanks for checking, just adjusted.

vaukai commented:
fop restricted to `<=virtual/jdk-17:*` because of trouble with `cannot implement removeLast()` when using `java 21`:

```
fop-core/src/main/java/org/apache/fop/layoutmgr/KnuthSequence.java:147: error: removeLast() in KnuthSequence cannot implement removeLast() in List
    public ListElement removeLast() {
                       ^
  return type ListElement is not compatible with T
  where E,T are type-variables:
    E extends Object declared in interface List
    T extends ListElement declared in class KnuthSequence
fop-core/src/main/java/org/apache/fop/layoutmgr/KnuthSequence.java:137: error: getLast() in KnuthSequence cannot implement getLast() in List
    public ListElement getLast() {
                       ^
  return type ListElement is not compatible with T
  where E,T are type-variables:
    E extends Object declared in interface List
    T extends ListElement declared in class KnuthSequence
fop-core/src/main/java/org/apache/fop/layoutmgr/BlockKnuthSequence.java:28: error: methods removeLast() from KnuthSequence and removeLast() from ArrayList are inherited with the same signature
public class BlockKnuthSequence extends KnuthSequence {
       ^
  where E is a type-variable:
    E extends Object declared in class ArrayList
fop-core/src/main/java/org/apache/fop/fo/FOTreeBuilder.java:181: warning: [removal] AccessController in java.security has been deprecated and marked for removal
        SAXException saxException = AccessController.doPrivileged(
                                    ^
fop-core/src/main/java/org/apache/fop/fo/FOTreeBuilder.java:203: warning: [removal] AccessController in java.security has been deprecated and marked for removal
            SAXException saxException = AccessController.doPrivileged(
                                        ^
fop-core/src/main/java/org/apache/fop/cli/Main.java:123: warning: [removal] AccessController in java.security has been deprecated and marked for removal
                AccessController.doPrivileged(new PrivilegedAction() {
                ^
fop-core/src/main/java/org/apache/fop/tools/anttasks/RunTest.java:145: warning: [removal] AccessController in java.security has been deprecated and marked for removal
                AccessController.doPrivileged(new PrivilegedAction() {
                ^
fop-core/src/main/java/org/apache/fop/layoutmgr/InlineKnuthSequence.java:33: error: methods removeLast() from KnuthSequence and removeLast() from ArrayList are inherited with the same signature
public class InlineKnuthSequence<T extends ListElement> extends KnuthSequence  {
       ^
  where E is a type-variable:
    E extends Object declared in class ArrayList
fop-core/src/main/java/org/apache/fop/layoutmgr/inline/LineLayoutManager.java:196: error: methods removeLast() from KnuthSequence and removeLast() from ArrayList are inherited with the same signature
    private static class Paragraph extends InlineKnuthSequence {
                   ^
  where E is a type-variable:
    E extends Object declared in class ArrayList
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
5 errors
```

fordfrog reviewed (commented):
i can't get this compiled:
```
>>> Emerging (1 of 1) dev-java/commons-validator-1.9.0::gentoo
 * commons-validator-1.9.0-src.tar.gz BLAKE2B SHA512 size ;-) ...                                                                                                                                                                                                                                                                                                   [ ok ]
 * commons-validator-1.9.0-src.tar.gz.asc BLAKE2B SHA512 size ;-) ...                                                                                                                                                                                                                                                                                               [ ok ]
 * Using: openjdk-21
>>> Unpacking source...
 * Verifying commons-validator-1.9.0-src.tar.gz ...
INFO     File /var/tmp/portage/dev-java/commons-validator-1.9.0/distdir/commons-validator-1.9.0-src.tar.gz verified successfully against the signature in /var/tmp/portage/dev-java/commons-validator-1.9.0/distdir/commons-validator-1.9.0-src.tar.gz.asc:                                                                                                                
INFO     - status: OpenPGPSignatureStatus.GOOD                                                                                                                                                                                                                                                                                                                             
INFO     - valid: True, trusted: True                                                                                                                                                                                                                                                                                                                                      
INFO     - primary key: 2DB4F1EF0FA761ECC4EA935C86FDC7E2A11262CB                                                                                                                                                                                                                                                                                                           
INFO     - subkey: 2DB4F1EF0FA761ECC4EA935C86FDC7E2A11262CB                                                                                                                                                                                                                                                                                                                
INFO     - timestamp: 2024-05-25 20:36:23 UTC                                                                                                                                                                                                                                                                                                                              
>>> Unpacking commons-validator-1.9.0-src.tar.gz to /var/tmp/portage/dev-java/commons-validator-1.9.0/work
=== Skipping unpack of commons-validator-1.9.0-src.tar.gz.asc
>>> Source unpacked in /var/tmp/portage/dev-java/commons-validator-1.9.0/work
>>> Preparing source in /var/tmp/portage/dev-java/commons-validator-1.9.0/work/commons-validator-1.9.0-src ...
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/dev-java/commons-validator-1.9.0/work/commons-validator-1.9.0-src ...
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-java/commons-validator-1.9.0/work/commons-validator-1.9.0-src ...
javac -source 1.8 -target 1.8 -d target/classes -encoding UTF-8 -classpath /usr/share/commons-beanutils-1.7/lib/commons-beanutils.jar:/usr/share/commons-digester-2.1/lib/commons-digester.jar:/usr/share/commons-logging/lib/commons-logging.jar:/usr/share/jakarta-servlet-api-4/lib/jakarta-servlet-api.jar:/usr/share/slf4j-api/lib/slf4j-api.jar:/usr/share/log4j-12-api-2/lib/log4j-12-api.jar:/usr/share/log4j-api-2/lib/log4j-api.jar:/usr/share/commons-collections/lib/commons-collections.jar:/usr/share/log4j-core-2/lib/log4j-core.jar:/usr/share/commons-compress/lib/commons-compress.jar:/usr/share/commons-csv/lib/commons-csv.jar:/usr/share/conversant-disruptor/lib/conversant-disruptor.jar:/usr/share/jackson-core/lib/jackson-core.jar:/usr/share/jackson-databind/lib/jackson-databind.jar:/usr/share/jackson-dataformat-xml/lib/jackson-dataformat-xml.jar:/usr/share/jackson-dataformat-yaml/lib/jackson-dataformat-yaml.jar:/usr/share/jansi-2/lib/jansi.jar:/usr/share/javax-mail/lib/javax-mail.jar:/usr/share/jctools-core/lib/jctools-core.jar:/usr/share/jeromq/lib/jeromq.jar:/usr/share/kafka-clients/lib/kafka-clients.jar:/usr/share/lmax-disruptor/lib/lmax-disruptor.jar:/usr/share/woodstox-core/lib/woodstox-core.jar:/usr/share/jackson-annotations-2/lib/jackson-annotations.jar:/usr/share/stax2-api/lib/stax2-api.jar:/usr/share/lz4-java/lib/lz4-java.jar:/usr/share/snappy-java/lib/snappy-java.jar:/usr/share/msv/lib/msv.jar:/usr/share/relaxng-datatype/lib/relaxng-datatype.jar:/usr/share/xsdlib/lib/xsdlib.jar:/usr/share/osgi-core/lib/osgi-core.jar:/usr/share/asm-9/lib/asm.jar:/usr/share/asm-9/lib/asm-analysis.jar:/usr/share/asm-9/lib/asm-commons.jar:/usr/share/asm-9/lib/asm-tree.jar:/usr/share/asm-9/lib/asm-util.jar:/usr/share/brotli-dec/lib/brotli-dec.jar:/usr/share/commons-codec/lib/commons-codec.jar:/usr/share/commons-io-1/lib/commons-io.jar:/usr/share/commons-lang-3.6/lib/commons-lang.jar:/usr/share/xz-java/lib/xz-java.jar:/usr/share/zstd-jni/lib/zstd-jni.jar:/usr/share/xerces-2/lib/xerces.jar:/usr/share/iso-relax/lib/iso-relax.jar:/usr/share/xml-commons-external-1.4/lib/xml-commons-external.jar:/usr/share/xml-commons-resolver/lib/xml-commons-resolver.jar:/usr/share/ant/lib/ant.jar:/usr/share/ant/lib/ant-launcher.jar:/usr/share/ant/lib/ant-apache-bsf.jar:/usr/share/ant/lib/ant-junit.jar:/usr/share/ant/lib/ant-junit4.jar @sources.lst
warning: [options] bootstrap class path not set in conjunction with -source 8
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
Note: Annotation processing is enabled because one or more processors were found
  on the class path. A future release of javac may disable annotation processing
  unless at least one processor is specified by name (-processor), or a search
  path is specified (--processor-path, --processor-module-path), or annotation
  processing is enabled explicitly (-proc:only, -proc:full).
  Use -Xlint:-options to suppress this message.
  Use -proc:none to disable annotation processing.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
4 warnings
Exception in thread "main" java.lang.module.FindException: Module log4j.api not found, required by org.apache.commons.logging
	at java.base/java.lang.module.Resolver.findFail(Resolver.java:892)
	at java.base/java.lang.module.Resolver.resolve(Resolver.java:192)
	at java.base/java.lang.module.Resolver.resolve(Resolver.java:141)
	at java.base/java.lang.module.Configuration.resolve(Configuration.java:420)
	at java.base/java.lang.module.Configuration.resolve(Configuration.java:254)
	at jdk.jdeps/com.sun.tools.jdeps.JdepsConfiguration$Builder.build(JdepsConfiguration.java:564)
	at jdk.jdeps/com.sun.tools.jdeps.JdepsTask.buildConfig(JdepsTask.java:607)
	at jdk.jdeps/com.sun.tools.jdeps.JdepsTask.run(JdepsTask.java:561)
	at jdk.jdeps/com.sun.tools.jdeps.JdepsTask.run(JdepsTask.java:537)
	at jdk.jdeps/com.sun.tools.jdeps.Main.main(Main.java:50)
 * ERROR: dev-java/commons-validator-1.9.0::gentoo failed (compile phase):
 *   (no error message)
 * 
 * Call stack:
 *     ebuild.sh, line 136:  Called src_compile
 *   environment, line 2409:  Called die
 * The specific snippet of code:
 *       jdeps --module-path "${modulepath}" --add-modules=ALL-MODULE-PATH --ignore-missing-deps --generate-module-info src/main --multi-release 9 "${JAVA_MODULE_NAME}.jar" || die;
```

vaukai commented:
> i can't get this compiled:
> 
> ```
> >>> Emerging (1 of 1) dev-java/commons-validator-1.9.0::gentoo
>  * commons-validator-1.9.0-src.tar.gz BLAKE2B SHA512 size ;-) ...                                                                                                                                                                                                                                                                                                   [ ok ]
>  * commons-validator-1.9.0-src.tar.gz.asc BLAKE2B SHA512 size ;-) ...                                                                                                                                                                                                                                                                                               [ ok ]
>  * Using: openjdk-21
> >>> Unpacking source...
>  * Verifying commons-validator-1.9.0-src.tar.gz ...
> INFO     File /var/tmp/portage/dev-java/commons-validator-1.9.0/distdir/commons-validator-1.9.0-src.tar.gz verified successfully against the signature in /var/tmp/portage/dev-java/commons-validator-1.9.0/distdir/commons-validator-1.9.0-src.tar.gz.asc:                                                                                                                
> INFO     - status: OpenPGPSignatureStatus.GOOD                                                                                                                                                                                                                                                                                                                             
> INFO     - valid: True, trusted: True                                                                                                                                                                                                                                                                                                                                      
> INFO     - primary key: 2DB4F1EF0FA761ECC4EA935C86FDC7E2A11262CB                                                                                                                                                                                                                                                                                                           
> INFO     - subkey: 2DB4F1EF0FA761ECC4EA935C86FDC7E2A11262CB                                                                                                                                                                                                                                                                                                                
> INFO     - timestamp: 2024-05-25 20:36:23 UTC                                                                                                                                                                                                                                                                                                                              
> >>> Unpacking commons-validator-1.9.0-src.tar.gz to /var/tmp/portage/dev-java/commons-validator-1.9.0/work
> === Skipping unpack of commons-validator-1.9.0-src.tar.gz.asc
> >>> Source unpacked in /var/tmp/portage/dev-java/commons-validator-1.9.0/work
> >>> Preparing source in /var/tmp/portage/dev-java/commons-validator-1.9.0/work/commons-validator-1.9.0-src ...
> >>> Source prepared.
> >>> Configuring source in /var/tmp/portage/dev-java/commons-validator-1.9.0/work/commons-validator-1.9.0-src ...
> >>> Source configured.
> >>> Compiling source in /var/tmp/portage/dev-java/commons-validator-1.9.0/work/commons-validator-1.9.0-src ...
> javac -source 1.8 -target 1.8 -d target/classes -encoding UTF-8 -classpath /usr/share/commons-beanutils-1.7/lib/commons-beanutils.jar:/usr/share/commons-digester-2.1/lib/commons-digester.jar:/usr/share/commons-logging/lib/commons-logging.jar:/usr/share/jakarta-servlet-api-4/lib/jakarta-servlet-api.jar:/usr/share/slf4j-api/lib/slf4j-api.jar:/usr/share/log4j-12-api-2/lib/log4j-12-api.jar:/usr/share/log4j-api-2/lib/log4j-api.jar:/usr/share/commons-collections/lib/commons-collections.jar:/usr/share/log4j-core-2/lib/log4j-core.jar:/usr/share/commons-compress/lib/commons-compress.jar:/usr/share/commons-csv/lib/commons-csv.jar:/usr/share/conversant-disruptor/lib/conversant-disruptor.jar:/usr/share/jackson-core/lib/jackson-core.jar:/usr/share/jackson-databind/lib/jackson-databind.jar:/usr/share/jackson-dataformat-xml/lib/jackson-dataformat-xml.jar:/usr/share/jackson-dataformat-yaml/lib/jackson-dataformat-yaml.jar:/usr/share/jansi-2/lib/jansi.jar:/usr/share/javax-mail/lib/javax-mail.jar:/usr/share/jctools-core/lib/jctools-core.jar:/usr/share/jeromq/lib/jeromq.jar:/usr/share/kafka-clients/lib/kafka-clients.jar:/usr/share/lmax-disruptor/lib/lmax-disruptor.jar:/usr/share/woodstox-core/lib/woodstox-core.jar:/usr/share/jackson-annotations-2/lib/jackson-annotations.jar:/usr/share/stax2-api/lib/stax2-api.jar:/usr/share/lz4-java/lib/lz4-java.jar:/usr/share/snappy-java/lib/snappy-java.jar:/usr/share/msv/lib/msv.jar:/usr/share/relaxng-datatype/lib/relaxng-datatype.jar:/usr/share/xsdlib/lib/xsdlib.jar:/usr/share/osgi-core/lib/osgi-core.jar:/usr/share/asm-9/lib/asm.jar:/usr/share/asm-9/lib/asm-analysis.jar:/usr/share/asm-9/lib/asm-commons.jar:/usr/share/asm-9/lib/asm-tree.jar:/usr/share/asm-9/lib/asm-util.jar:/usr/share/brotli-dec/lib/brotli-dec.jar:/usr/share/commons-codec/lib/commons-codec.jar:/usr/share/commons-io-1/lib/commons-io.jar:/usr/share/commons-lang-3.6/lib/commons-lang.jar:/usr/share/xz-java/lib/xz-java.jar:/usr/share/zstd-jni/lib/zstd-jni.jar:/usr/share/xerces-2/lib/xerces.jar:/usr/share/iso-relax/lib/iso-relax.jar:/usr/share/xml-commons-external-1.4/lib/xml-commons-external.jar:/usr/share/xml-commons-resolver/lib/xml-commons-resolver.jar:/usr/share/ant/lib/ant.jar:/usr/share/ant/lib/ant-launcher.jar:/usr/share/ant/lib/ant-apache-bsf.jar:/usr/share/ant/lib/ant-junit.jar:/usr/share/ant/lib/ant-junit4.jar @sources.lst
> warning: [options] bootstrap class path not set in conjunction with -source 8
> warning: [options] source value 8 is obsolete and will be removed in a future release
> warning: [options] target value 8 is obsolete and will be removed in a future release
> warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
> Note: Annotation processing is enabled because one or more processors were found
>   on the class path. A future release of javac may disable annotation processing
>   unless at least one processor is specified by name (-processor), or a search
>   path is specified (--processor-path, --processor-module-path), or annotation
>   processing is enabled explicitly (-proc:only, -proc:full).
>   Use -Xlint:-options to suppress this message.
>   Use -proc:none to disable annotation processing.
> Note: Some input files use or override a deprecated API.
> Note: Recompile with -Xlint:deprecation for details.
> 4 warnings
> Exception in thread "main" java.lang.module.FindException: Module log4j.api not found, required by org.apache.commons.logging
> 	at java.base/java.lang.module.Resolver.findFail(Resolver.java:892)
> 	at java.base/java.lang.module.Resolver.resolve(Resolver.java:192)
> 	at java.base/java.lang.module.Resolver.resolve(Resolver.java:141)
> 	at java.base/java.lang.module.Configuration.resolve(Configuration.java:420)
> 	at java.base/java.lang.module.Configuration.resolve(Configuration.java:254)
> 	at jdk.jdeps/com.sun.tools.jdeps.JdepsConfiguration$Builder.build(JdepsConfiguration.java:564)
> 	at jdk.jdeps/com.sun.tools.jdeps.JdepsTask.buildConfig(JdepsTask.java:607)
> 	at jdk.jdeps/com.sun.tools.jdeps.JdepsTask.run(JdepsTask.java:561)
> 	at jdk.jdeps/com.sun.tools.jdeps.JdepsTask.run(JdepsTask.java:537)
> 	at jdk.jdeps/com.sun.tools.jdeps.Main.main(Main.java:50)
>  * ERROR: dev-java/commons-validator-1.9.0::gentoo failed (compile phase):
>  *   (no error message)
>  * 
>  * Call stack:
>  *     ebuild.sh, line 136:  Called src_compile
>  *   environment, line 2409:  Called die
>  * The specific snippet of code:
>  *       jdeps --module-path "${modulepath}" --add-modules=ALL-MODULE-PATH --ignore-missing-deps --generate-module-info src/main --multi-release 9 "${JAVA_MODULE_NAME}.jar" || die;
> ```


finding the root cause of an error is sometimes difficult, especially when i cannot reproduce it. 
you seem to be having `ECLASS_DEBUG_OUTPUT=off`. could you please turn it on and try again?
for me it compiles without any trouble. build log is here:
```
[32m * [39;49;00mPackage:    dev-java/commons-validator-1.9.0:0
[32m * [39;49;00mRepository: gentoo
[32m * [39;49;00mMaintainer: java@gentoo.org
[32m * [39;49;00mUpstream:   https://issues.apache.org/jira/projects/VALIDATOR/issues/
[32m * [39;49;00mUSE:        abi_x86_64 amd64 elibc_glibc kernel_linux
[32m * [39;49;00mFEATURES:   network-sandbox preserve-libs sandbox userpriv usersandbox
debug:   eclass exists: /var/db/repos/vaukai/eclass/java-pkg-2.eclass
debug: inherit: java-pkg-2 -> /var/db/repos/vaukai/eclass/java-pkg-2.eclass
debug: *** Multiple Inheritance (Level: 2)
debug:   eclass exists: /var/db/repos/vaukai/eclass/java-utils-2.eclass
debug: inherit: java-utils-2 -> /var/db/repos/vaukai/eclass/java-utils-2.eclass
debug: EXPORT_FUNCTIONS: pkg_setup -> java-pkg-2_pkg_setup
debug: EXPORT_FUNCTIONS: src_prepare -> java-pkg-2_src_prepare
debug: EXPORT_FUNCTIONS: pkg_preinst -> java-pkg-2_pkg_preinst
debug:   eclass exists: /var/db/repos/vaukai/eclass/java-pkg-simple.eclass
debug: inherit: java-pkg-simple -> /var/db/repos/vaukai/eclass/java-pkg-simple.eclass
debug: *** Multiple Inheritance (Level: 2)
debug:   eclass exists: /var/db/repos/vaukai/eclass/java-utils-2.eclass
debug: inherit: java-utils-2 -> /var/db/repos/vaukai/eclass/java-utils-2.eclass
debug: EXPORT_FUNCTIONS: src_compile -> java-pkg-simple_src_compile
debug: EXPORT_FUNCTIONS: src_install -> java-pkg-simple_src_install
debug: EXPORT_FUNCTIONS: src_test -> java-pkg-simple_src_test
debug:   eclass exists: /var/db/repos/vaukai/eclass/verify-sig.eclass
debug: inherit: verify-sig -> /var/db/repos/vaukai/eclass/verify-sig.eclass
debug: EXPORT_FUNCTIONS: src_unpack -> verify-sig_src_unpack
[32m * [39;49;00mPackage:    dev-java/commons-validator-1.9.0:0
[32m * [39;49;00mRepository: gentoo
[32m * [39;49;00mMaintainer: java@gentoo.org
[32m * [39;49;00mUpstream:   https://issues.apache.org/jira/projects/VALIDATOR/issues/
[32m * [39;49;00mUSE:        abi_x86_64 amd64 elibc_glibc kernel_linux
[32m * [39;49;00mFEATURES:   network-sandbox preserve-libs sandbox userpriv usersandbox
debug:   eclass exists: /var/db/repos/vaukai/eclass/java-pkg-2.eclass
debug: inherit: java-pkg-2 -> /var/db/repos/vaukai/eclass/java-pkg-2.eclass
debug: *** Multiple Inheritance (Level: 2)
debug:   eclass exists: /var/db/repos/vaukai/eclass/java-utils-2.eclass
debug: inherit: java-utils-2 -> /var/db/repos/vaukai/eclass/java-utils-2.eclass
debug: EXPORT_FUNCTIONS: pkg_setup -> java-pkg-2_pkg_setup
debug: EXPORT_FUNCTIONS: src_prepare -> java-pkg-2_src_prepare
debug: EXPORT_FUNCTIONS: pkg_preinst -> java-pkg-2_pkg_preinst
debug:   eclass exists: /var/db/repos/vaukai/eclass/java-pkg-simple.eclass
debug: inherit: java-pkg-simple -> /var/db/repos/vaukai/eclass/java-pkg-simple.eclass
debug: *** Multiple Inheritance (Level: 2)
debug:   eclass exists: /var/db/repos/vaukai/eclass/java-utils-2.eclass
debug: inherit: java-utils-2 -> /var/db/repos/vaukai/eclass/java-utils-2.eclass
debug: EXPORT_FUNCTIONS: src_compile -> java-pkg-simple_src_compile
debug: EXPORT_FUNCTIONS: src_install -> java-pkg-simple_src_install
debug: EXPORT_FUNCTIONS: src_test -> java-pkg-simple_src_test
debug:   eclass exists: /var/db/repos/vaukai/eclass/verify-sig.eclass
debug: inherit: verify-sig -> /var/db/repos/vaukai/eclass/verify-sig.eclass
debug: EXPORT_FUNCTIONS: src_unpack -> verify-sig_src_unpack
debug: java-pkg_init: entering function, parameters: 
debug: java-pkg_switch-vm: entering function, parameters: 
debug: java-pkg_needs-vm: entering function, parameters: 
debug: depend-java-query:  NV_DEPEND:	
	
	dev-java/commons-beanutils:1.7
	dev-java/commons-digester:2.1
	dev-java/commons-logging:0
	dev-java/jakarta-servlet-api:4
	dev-java/slf4j-api:0

	>=virtual/jdk-1.8:*
 >=dev-java/java-config-2.2.0-r3 source? ( app-arch/zip ) test? (  dev-java/junit:4 )
debug: Using: openjdk-21
debug: java-pkg_setup-vm: entering function, parameters: 
debug: java-pkg_get-vm-vendor: entering function, parameters: 
debug: java-pkg_javac-args: entering function, parameters: 
debug: want source: 1.8
debug: want target: 1.8
debug: java-pkg_append_: entering function, parameters: LD_LIBRARY_PATH /usr/lib64/openjdk-21/lib/:/usr/lib64/openjdk-21/lib/server/
 [32m*[0m Using: openjdk-21
>>> Unpacking source...
>>> Unpacking commons-validator-1.9.0-src.tar.gz to /var/tmp/portage/dev-java/commons-validator-1.9.0/work
>>> Source unpacked in /var/tmp/portage/dev-java/commons-validator-1.9.0/work
>>> Preparing source in /var/tmp/portage/dev-java/commons-validator-1.9.0/work/commons-validator-1.9.0-src ...
Searching for bundled jars:
None found.
Searching for bundled classes (no output if none found):
Search done.
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/dev-java/commons-validator-1.9.0/work/commons-validator-1.9.0-src ...
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-java/commons-validator-1.9.0/work/commons-validator-1.9.0-src ...
debug: java-pkg_getjars: entering function, parameters: --build-only commons-beanutils-1.7
debug: commons-beanutils-1.7:/usr/share/commons-beanutils-1.7/lib/commons-beanutils.jar
debug: java-pkg_ensure-dep: entering function, parameters: build commons-beanutils-1.7
debug: Matching against: /commons-beanutils(-1\.7(-[^:]+)?:0|(-[^:]+)?:1\.7)\b
debug: java-pkg_record-jar_: entering function, parameters: --build-only commons-beanutils-1.7
debug: java-pkg_getjars: entering function, parameters: --build-only commons-digester-2.1
debug: commons-digester-2.1:/usr/share/commons-digester-2.1/lib/commons-digester.jar
debug: java-pkg_ensure-dep: entering function, parameters: build commons-digester-2.1
debug: Matching against: /commons-digester(-2\.1(-[^:]+)?:0|(-[^:]+)?:2\.1)\b
debug: java-pkg_record-jar_: entering function, parameters: --build-only commons-digester-2.1
debug: java-pkg_getjars: entering function, parameters: --build-only commons-logging
debug: commons-logging:/usr/share/commons-logging/lib/commons-logging.jar
debug: java-pkg_ensure-dep: entering function, parameters: build commons-logging
debug: Matching against: /commons(-logging(-[^:]+)?:0|(-[^:]+)?:logging)\b
debug: java-pkg_record-jar_: entering function, parameters: --build-only commons-logging
debug: java-pkg_getjars: entering function, parameters: --build-only jakarta-servlet-api-4
debug: jakarta-servlet-api-4:/usr/share/jakarta-servlet-api-4/lib/jakarta-servlet-api.jar
debug: java-pkg_ensure-dep: entering function, parameters: build jakarta-servlet-api-4
debug: Matching against: /jakarta-servlet-api(-4(-[^:]+)?:0|(-[^:]+)?:4)\b
debug: java-pkg_record-jar_: entering function, parameters: --build-only jakarta-servlet-api-4
debug: java-pkg_getjars: entering function, parameters: --build-only slf4j-api
debug: slf4j-api:/usr/share/slf4j-api/lib/slf4j-api.jar
debug: java-pkg_ensure-dep: entering function, parameters: build slf4j-api
debug: Matching against: /slf4j(-api(-[^:]+)?:0|(-[^:]+)?:api)\b
debug: java-pkg_record-jar_: entering function, parameters: --build-only slf4j-api
debug: java-pkg_gen-cp: entering function, parameters: JAVA_GENTOO_CLASSPATH
debug: java-pkg-simple_getclasspath: entering function, parameters: 
debug: java-pkg_getjars: entering function, parameters: --with-dependencies commons-beanutils-1.7,commons-digester-2.1,commons-logging,jakarta-servlet-api-4,slf4j-api
debug: commons-beanutils-1.7,commons-digester-2.1,commons-logging,jakarta-servlet-api-4,slf4j-api:/usr/share/commons-beanutils-1.7/lib/commons-beanutils.jar:/usr/share/commons-digester-2.1/lib/commons-digester.jar:/usr/share/commons-logging/lib/commons-logging.jar:/usr/share/jakarta-servlet-api-4/lib/jakarta-servlet-api.jar:/usr/share/slf4j-api/lib/slf4j-api.jar:/usr/share/commons-collections/lib/commons-collections.jar
debug: java-pkg_ensure-dep: entering function, parameters: commons-beanutils-1.7
debug: Matching against: /commons-beanutils(-1\.7(-[^:]+)?:0|(-[^:]+)?:1\.7)\b
debug: java-pkg_ensure-dep: entering function, parameters: commons-digester-2.1
debug: Matching against: /commons-digester(-2\.1(-[^:]+)?:0|(-[^:]+)?:2\.1)\b
debug: java-pkg_ensure-dep: entering function, parameters: commons-logging
debug: Matching against: /commons(-logging(-[^:]+)?:0|(-[^:]+)?:logging)\b
debug: java-pkg_ensure-dep: entering function, parameters: jakarta-servlet-api-4
debug: Matching against: /jakarta-servlet-api(-4(-[^:]+)?:0|(-[^:]+)?:4)\b
debug: java-pkg_ensure-dep: entering function, parameters: slf4j-api
debug: Matching against: /slf4j(-api(-[^:]+)?:0|(-[^:]+)?:api)\b
debug: java-pkg_record-jar_: entering function, parameters: commons-beanutils-1.7
debug: java-pkg_record-jar_: entering function, parameters: commons-digester-2.1
debug: java-pkg_record-jar_: entering function, parameters: commons-logging
debug: java-pkg_record-jar_: entering function, parameters: jakarta-servlet-api-4
debug: java-pkg_record-jar_: entering function, parameters: slf4j-api
debug: CLASSPATH=/usr/share/commons-beanutils-1.7/lib/commons-beanutils.jar:/usr/share/commons-digester-2.1/lib/commons-digester.jar:/usr/share/commons-logging/lib/commons-logging.jar:/usr/share/jakarta-servlet-api-4/lib/jakarta-servlet-api.jar:/usr/share/slf4j-api/lib/slf4j-api.jar:/usr/share/commons-collections/lib/commons-collections.jar
debug: java-pkg-simple_prepend_resources: entering function, parameters: target/classes src/main/resources
debug: ejavac: entering function, parameters: -d target/classes -encoding UTF-8 -classpath /usr/share/commons-beanutils-1.7/lib/commons-beanutils.jar:/usr/share/commons-digester-2.1/lib/commons-digester.jar:/usr/share/commons-logging/lib/commons-logging.jar:/usr/share/jakarta-servlet-api-4/lib/jakarta-servlet-api.jar:/usr/share/slf4j-api/lib/slf4j-api.jar:/usr/share/commons-collections/lib/commons-collections.jar @sources.lst
debug: java-pkg_get-javac: entering function, parameters: 
debug: java-pkg_init-compiler_: entering function, parameters: 
debug: Read "" from /etc/java-config-2/build/compilers.conf
debug: java-pkg_javac-args: entering function, parameters: 
debug: want source: 1.8
debug: want target: 1.8
 [32m*[0m Verbose logging for "ejavac" function
 [32m*[0m Compiler executable: javac
 [32m*[0m Extra arguments: -source 1.8 -target 1.8
 [32m*[0m Complete command:
 [32m*[0m javac -source 1.8 -target 1.8 -d target/classes -encoding UTF-8 -classpath /usr/share/commons-beanutils-1.7/lib/commons-beanutils.jar:/usr/share/commons-digester-2.1/lib/commons-digester.jar:/usr/share/commons-logging/lib/commons-logging.jar:/usr/share/jakarta-servlet-api-4/lib/jakarta-servlet-api.jar:/usr/share/slf4j-api/lib/slf4j-api.jar:/usr/share/commons-collections/lib/commons-collections.jar @sources.lst
javac -source 1.8 -target 1.8 -d target/classes -encoding UTF-8 -classpath /usr/share/commons-beanutils-1.7/lib/commons-beanutils.jar:/usr/share/commons-digester-2.1/lib/commons-digester.jar:/usr/share/commons-logging/lib/commons-logging.jar:/usr/share/jakarta-servlet-api-4/lib/jakarta-servlet-api.jar:/usr/share/slf4j-api/lib/slf4j-api.jar:/usr/share/commons-collections/lib/commons-collections.jar @sources.lst
warning: [options] bootstrap class path not set in conjunction with -source 8
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
4 warnings
Warning: split package: org.apache.commons.validator file:///var/tmp/portage/dev-java/commons-validator-1.9.0/work/commons-validator-1.9.0-src/org.apache.commons.validator.jar org.apache.commons.validator.jar
Warning: split package: org.apache.commons.validator.routines file:///var/tmp/portage/dev-java/commons-validator-1.9.0/work/commons-validator-1.9.0-src/org.apache.commons.validator.jar org.apache.commons.validator.jar
Warning: split package: org.apache.commons.validator.routines.checkdigit file:///var/tmp/portage/dev-java/commons-validator-1.9.0/work/commons-validator-1.9.0-src/org.apache.commons.validator.jar org.apache.commons.validator.jar
Warning: split package: org.apache.commons.validator.util file:///var/tmp/portage/dev-java/commons-validator-1.9.0/work/commons-validator-1.9.0-src/org.apache.commons.validator.jar org.apache.commons.validator.jar
Warning: --ignore-missing-deps specified. Missing dependencies from org.apache.commons.validator are ignored
writing to src/main/org.apache.commons.validator/versions/9/module-info.java
debug: ejavac: entering function, parameters: -source 9 -target 9 --module-path :/usr/share/commons-beanutils-1.7/lib/commons-beanutils.jar:/usr/share/commons-digester-2.1/lib/commons-digester.jar:/usr/share/commons-logging/lib/commons-logging.jar:/usr/share/jakarta-servlet-api-4/lib/jakarta-servlet-api.jar:/usr/share/slf4j-api/lib/slf4j-api.jar --patch-module org.apache.commons.validator=org.apache.commons.validator.jar -d target/versions/9 src/main/org.apache.commons.validator/versions/9/module-info.java
debug: java-pkg_get-javac: entering function, parameters: 
debug: java-pkg_init-compiler_: entering function, parameters: 
debug: Read "" from /etc/java-config-2/build/compilers.conf
debug: java-pkg_javac-args: entering function, parameters: 
debug: want source: 1.8
debug: want target: 1.8
 [32m*[0m Verbose logging for "ejavac" function
 [32m*[0m Compiler executable: javac
 [32m*[0m Extra arguments: -source 1.8 -target 1.8
 [32m*[0m Complete command:
 [32m*[0m javac -source 1.8 -target 1.8 -source 9 -target 9 --module-path :/usr/share/commons-beanutils-1.7/lib/commons-beanutils.jar:/usr/share/commons-digester-2.1/lib/commons-digester.jar:/usr/share/commons-logging/lib/commons-logging.jar:/usr/share/jakarta-servlet-api-4/lib/jakarta-servlet-api.jar:/usr/share/slf4j-api/lib/slf4j-api.jar --patch-module org.apache.commons.validator=org.apache.commons.validator.jar -d target/versions/9 src/main/org.apache.commons.validator/versions/9/module-info.java
javac -source 1.8 -target 1.8 -source 9 -target 9 --module-path :/usr/share/commons-beanutils-1.7/lib/commons-beanutils.jar:/usr/share/commons-digester-2.1/lib/commons-digester.jar:/usr/share/commons-logging/lib/commons-logging.jar:/usr/share/jakarta-servlet-api-4/lib/jakarta-servlet-api.jar:/usr/share/slf4j-api/lib/slf4j-api.jar --patch-module org.apache.commons.validator=org.apache.commons.validator.jar -d target/versions/9 src/main/org.apache.commons.validator/versions/9/module-info.java
warning: [options] system modules path not set in conjunction with -source 9
src/main/org.apache.commons.validator/versions/9/module-info.java:5: warning: requires transitive directive for an automatic module
    requires transitive commons.digester;
                               ^
2 warnings
added manifest
added module-info: META-INF/versions/9/module-info.class
adding: org/(in = 0) (out= 0)(stored 0%)
adding: org/apache/(in = 0) (out= 0)(stored 0%)
adding: org/apache/commons/(in = 0) (out= 0)(stored 0%)
adding: org/apache/commons/validator/(in = 0) (out= 0)(stored 0%)
adding: org/apache/commons/validator/Arg.class(in = 1897) (out= 971)(deflated 48%)
adding: org/apache/commons/validator/CreditCardValidator$Amex.class(in = 1114) (out= 613)(deflated 44%)
adding: org/apache/commons/validator/CreditCardValidator$CreditCardType.class(in = 307) (out= 199)(deflated 35%)
adding: org/apache/commons/validator/CreditCardValidator$Discover.class(in = 935) (out= 517)(deflated 44%)
adding: org/apache/commons/validator/CreditCardValidator$Mastercard.class(in = 1141) (out= 624)(deflated 45%)
adding: org/apache/commons/validator/CreditCardValidator$Visa.class(in = 930) (out= 514)(deflated 44%)
adding: org/apache/commons/validator/CreditCardValidator.class(in = 3058) (out= 1547)(deflated 49%)
adding: org/apache/commons/validator/DateValidator.class(in = 1467) (out= 854)(deflated 41%)
adding: org/apache/commons/validator/EmailValidator.class(in = 3534) (out= 1837)(deflated 48%)
adding: org/apache/commons/validator/Field.class(in = 14324) (out= 6362)(deflated 55%)
adding: org/apache/commons/validator/Form.class(in = 5623) (out= 2424)(deflated 56%)
adding: org/apache/commons/validator/FormSet.class(in = 5220) (out= 2541)(deflated 51%)
adding: org/apache/commons/validator/FormSetFactory.class(in = 2233) (out= 1054)(deflated 52%)
adding: org/apache/commons/validator/GenericTypeValidator.class(in = 5933) (out= 2589)(deflated 56%)
adding: org/apache/commons/validator/GenericValidator.class(in = 4445) (out= 1719)(deflated 61%)
adding: org/apache/commons/validator/ISBNValidator.class(in = 562) (out= 346)(deflated 38%)
adding: org/apache/commons/validator/Msg.class(in = 1661) (out= 877)(deflated 47%)
adding: org/apache/commons/validator/UrlValidator.class(in = 5501) (out= 3080)(deflated 44%)
adding: org/apache/commons/validator/Validator.class(in = 4271) (out= 1832)(deflated 57%)
adding: org/apache/commons/validator/ValidatorAction.class(in = 12244) (out= 5580)(deflated 54%)
adding: org/apache/commons/validator/ValidatorException.class(in = 392) (out= 279)(deflated 28%)
adding: org/apache/commons/validator/ValidatorResources$1.class(in = 2184) (out= 1070)(deflated 51%)
adding: org/apache/commons/validator/ValidatorResources.class(in = 11044) (out= 4547)(deflated 58%)
adding: org/apache/commons/validator/ValidatorResult$ResultStatus.class(in = 1060) (out= 570)(deflated 46%)
adding: org/apache/commons/validator/ValidatorResult.class(in = 2268) (out= 1054)(deflated 53%)
adding: org/apache/commons/validator/ValidatorResults.class(in = 2573) (out= 1204)(deflated 53%)
adding: org/apache/commons/validator/Var.class(in = 2130) (out= 1079)(deflated 49%)
adding: org/apache/commons/validator/digester-rules.xml(in = 4365) (out= 1049)(deflated 75%)
adding: org/apache/commons/validator/resources/(in = 0) (out= 0)(stored 0%)
adding: org/apache/commons/validator/resources/validator_1_0.dtd(in = 9210) (out= 2432)(deflated 73%)
adding: org/apache/commons/validator/resources/validator_1_0_1.dtd(in = 9132) (out= 2407)(deflated 73%)
adding: org/apache/commons/validator/resources/validator_1_1.dtd(in = 10266) (out= 2610)(deflated 74%)
adding: org/apache/commons/validator/resources/validator_1_1_3.dtd(in = 11424) (out= 2911)(deflated 74%)
adding: org/apache/commons/validator/resources/validator_1_2_0.dtd(in = 9247) (out= 2933)(deflated 68%)
adding: org/apache/commons/validator/resources/validator_1_3_0.dtd(in = 9247) (out= 2939)(deflated 68%)
adding: org/apache/commons/validator/resources/validator_1_4_0.dtd(in = 9479) (out= 2986)(deflated 68%)
adding: org/apache/commons/validator/routines/(in = 0) (out= 0)(stored 0%)
adding: org/apache/commons/validator/routines/AbstractCalendarValidator.class(in = 4749) (out= 2254)(deflated 52%)
adding: org/apache/commons/validator/routines/AbstractFormatValidator.class(in = 2100) (out= 926)(deflated 55%)
adding: org/apache/commons/validator/routines/AbstractNumberValidator.class(in = 3383) (out= 1733)(deflated 48%)
adding: org/apache/commons/validator/routines/BigDecimalValidator.class(in = 2304) (out= 1030)(deflated 55%)
adding: org/apache/commons/validator/routines/BigIntegerValidator.class(in = 1903) (out= 807)(deflated 57%)
adding: org/apache/commons/validator/routines/ByteValidator.class(in = 2099) (out= 949)(deflated 54%)
adding: org/apache/commons/validator/routines/CalendarValidator.class(in = 3062) (out= 1145)(deflated 62%)
adding: org/apache/commons/validator/routines/CodeValidator.class(in = 2775) (out= 1084)(deflated 60%)
adding: org/apache/commons/validator/routines/CreditCardValidator$1.class(in = 1987) (out= 928)(deflated 53%)
adding: org/apache/commons/validator/routines/CreditCardValidator$CreditCardRange.class(in = 819) (out= 473)(deflated 42%)
adding: org/apache/commons/validator/routines/CreditCardValidator.class(in = 6284) (out= 2680)(deflated 57%)
adding: org/apache/commons/validator/routines/CurrencyValidator.class(in = 1565) (out= 907)(deflated 42%)
adding: org/apache/commons/validator/routines/DateValidator.class(in = 3075) (out= 1130)(deflated 63%)
adding: org/apache/commons/validator/routines/DomainValidator$1.class(in = 269) (out= 181)(deflated 32%)
adding: org/apache/commons/validator/routines/DomainValidator$ArrayType.class(in = 1821) (out= 865)(deflated 52%)
adding: org/apache/commons/validator/routines/DomainValidator$IDNBUGHOLDER.class(in = 767) (out= 475)(deflated 38%)
adding: org/apache/commons/validator/routines/DomainValidator$Item.class(in = 687) (out= 351)(deflated 48%)
adding: org/apache/commons/validator/routines/DomainValidator$LazyHolder.class(in = 890) (out= 437)(deflated 50%)
adding: org/apache/commons/validator/routines/DomainValidator.class(in = 36231) (out= 20588)(deflated 43%)
adding: org/apache/commons/validator/routines/DoubleValidator.class(in = 2064) (out= 902)(deflated 56%)
adding: org/apache/commons/validator/routines/EmailValidator.class(in = 3664) (out= 1808)(deflated 50%)
adding: org/apache/commons/validator/routines/FloatValidator.class(in = 2207) (out= 1018)(deflated 53%)
adding: org/apache/commons/validator/routines/IBANValidator$Validator.class(in = 2308) (out= 1231)(deflated 46%)
adding: org/apache/commons/validator/routines/IBANValidator.class(in = 8453) (out= 3685)(deflated 56%)
adding: org/apache/commons/validator/routines/ISBNValidator.class(in = 3728) (out= 1708)(deflated 54%)
adding: org/apache/commons/validator/routines/ISINValidator.class(in = 7543) (out= 4163)(deflated 44%)
adding: org/apache/commons/validator/routines/ISSNValidator.class(in = 3268) (out= 1566)(deflated 52%)
adding: org/apache/commons/validator/routines/InetAddressValidator.class(in = 3761) (out= 2127)(deflated 43%)
adding: org/apache/commons/validator/routines/IntegerValidator.class(in = 2133) (out= 954)(deflated 55%)
adding: org/apache/commons/validator/routines/LongValidator.class(in = 1960) (out= 860)(deflated 56%)
adding: org/apache/commons/validator/routines/PercentValidator.class(in = 1753) (out= 1015)(deflated 42%)
adding: org/apache/commons/validator/routines/RegexValidator.class(in = 2959) (out= 1615)(deflated 45%)
adding: org/apache/commons/validator/routines/ShortValidator.class(in = 2107) (out= 943)(deflated 55%)
adding: org/apache/commons/validator/routines/TimeValidator.class(in = 2480) (out= 874)(deflated 64%)
adding: org/apache/commons/validator/routines/UrlValidator.class(in = 7218) (out= 3688)(deflated 48%)
adding: org/apache/commons/validator/routines/checkdigit/(in = 0) (out= 0)(stored 0%)
adding: org/apache/commons/validator/routines/checkdigit/ABANumberCheckDigit.class(in = 690) (out= 444)(deflated 35%)
adding: org/apache/commons/validator/routines/checkdigit/AbstractCheckDigit.class(in = 326) (out= 221)(deflated 32%)
adding: org/apache/commons/validator/routines/checkdigit/CASNumberCheckDigit.class(in = 2285) (out= 1181)(deflated 48%)
adding: org/apache/commons/validator/routines/checkdigit/CUSIPCheckDigit.class(in = 1360) (out= 806)(deflated 40%)
adding: org/apache/commons/validator/routines/checkdigit/CheckDigit.class(in = 350) (out= 208)(deflated 40%)
adding: org/apache/commons/validator/routines/checkdigit/CheckDigitException.class(in = 518) (out= 339)(deflated 34%)
adding: org/apache/commons/validator/routines/checkdigit/EAN13CheckDigit.class(in = 670) (out= 432)(deflated 35%)
adding: org/apache/commons/validator/routines/checkdigit/ECNumberCheckDigit.class(in = 2152) (out= 1081)(deflated 49%)
adding: org/apache/commons/validator/routines/checkdigit/IBANCheckDigit.class(in = 2301) (out= 1308)(deflated 43%)
adding: org/apache/commons/validator/routines/checkdigit/ISBN10CheckDigit.class(in = 975) (out= 562)(deflated 42%)
adding: org/apache/commons/validator/routines/checkdigit/ISBNCheckDigit.class(in = 1794) (out= 881)(deflated 50%)
adding: org/apache/commons/validator/routines/checkdigit/ISINCheckDigit.class(in = 1734) (out= 1011)(deflated 41%)
adding: org/apache/commons/validator/routines/checkdigit/ISSNCheckDigit.class(in = 982) (out= 555)(deflated 43%)
adding: org/apache/commons/validator/routines/checkdigit/LuhnCheckDigit.class(in = 729) (out= 481)(deflated 34%)
adding: org/apache/commons/validator/routines/checkdigit/ModulusCheckDigit.class(in = 2472) (out= 1372)(deflated 44%)
adding: org/apache/commons/validator/routines/checkdigit/ModulusTenCheckDigit.class(in = 2194) (out= 1225)(deflated 44%)
adding: org/apache/commons/validator/routines/checkdigit/SedolCheckDigit.class(in = 1628) (out= 954)(deflated 41%)
adding: org/apache/commons/validator/routines/checkdigit/VerhoeffCheckDigit.class(in = 3003) (out= 1506)(deflated 49%)
adding: org/apache/commons/validator/util/(in = 0) (out= 0)(stored 0%)
adding: org/apache/commons/validator/util/Flags.class(in = 1876) (out= 1033)(deflated 44%)
adding: org/apache/commons/validator/util/ValidatorUtils.class(in = 3762) (out= 1761)(deflated 53%)
adding: META-INF/versions/9/(in = 0) (out= 0)(stored 0%)
>>> Source compiled.
```

fordfrog commented:
i think the problem is that my `dev-java/log4j-api` does not provide `module-info.class` whereas yours already does.

vaukai commented:
> i think the problem is that my `dev-java/log4j-api` does not provide `module-info.class` whereas yours already does.

that is not the case, my `log4j*` are all unmodified from `::gentoo`. but i was able to reproduce that error with
 `>=dev-java/commons-logging-1.3.4-r1:0[log4j]`.
in order to avoid adding `IUSE=log4j` here i'd prefer depending on `dev-java/commons-logging:0[-log4j]`. there is presently nothing which would need `commons-logging` with that flag turned on.

fordfrog commented:
> > i think the problem is that my `dev-java/log4j-api` does not provide `module-info.class` whereas yours already does.
> 
> that is not the case, my `log4j*` are all unmodified from `::gentoo`. but i was able to reproduce that error with `>=dev-java/commons-logging-1.3.4-r1:0[log4j]`. in order to avoid adding `IUSE=log4j` here i'd prefer depending on `dev-java/commons-logging:0[-log4j]`. there is presently nothing which would need `commons-logging` with that flag turned on.

wouldn't adding module info to `log4j-api` be a better solution?

vaukai commented:
> > > i think the problem is that my `dev-java/log4j-api` does not provide `module-info.class` whereas yours already does.
> > 
> > 
> > that is not the case, my `log4j*` are all unmodified from `::gentoo`. but i was able to reproduce that error with `>=dev-java/commons-logging-1.3.4-r1:0[log4j]`. in order to avoid adding `IUSE=log4j` here i'd prefer depending on `dev-java/commons-logging:0[-log4j]`. there is presently nothing which would need `commons-logging` with that flag turned on.
> 
> wouldn't adding module info to `log4j-api` be a better solution?

agreed - if it solves the issue. but i haven't yet found the time to test that. i am in doubt it would work without adding `log4j-api` (optionally) to dependencies.

fordfrog reviewed (commented):
shouldn't be `17.0.12_p7` gone instead of `17.0.12_p11`?

vaukai commented:
> shouldn't be `17.0.12_p7` gone instead of `17.0.12_p11`?

you are right, sorry

fordfrog commented on dev-java/batik/batik-1.18.ebuild:
```diff
@@ -0,0 +1,247 @@
+
+pkg_postinst() {
+	xdg_desktop_database_update
+	xdg_icon_cache_update
```
aren't those ran by default?

vaukai replied:
> aren't those ran by default?

apparently not. dropping those lines would lead to:
```
Updating .desktop files database ...
debug: gtk-update-icon-cache is not found
```

vaukai commented:
@fordfrog for `fop-2.10` i've put two jar archives into FILESDIR, these would again need to get hosted in devspace [like before](https://github.com/gentoo/gentoo/pull/38836/commits/800599c79116c8dfc7ad164eb42babae5be1b6e5#diff-8377dd9ff25d1e324660cd97e6b069f9d61990c542dd61a727fe1350a9e72f67R16), please.

vaukai commented:
removing `dev-java/commons-validator` from this PR. will add it to the eclass works.
